From d99d7e2cdb79288e2735b317119b92a02bba9a75 Mon Sep 17 00:00:00 2001
From: Weili Qian <qianweili@huawei.com>
Date: Wed, 9 Mar 2022 10:15:34 +0800
Subject: [PATCH] crypto: hisilicon/qm - remove redundant cache writeback
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: d99d7e2cdb79288e2735b317119b92a02bba9a75
Modified-by-SEL: No


driver inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4W3OQ

----------------------------------------------------------------------

Currently, the memory of the queue's sqe is freed when
the driver is removed, not the put queue. Therefore, it is only
necessary to write back the data in the hardware cache to
memory before removing the driver.

Signed-off-by: Weili Qian <qianweili@huawei.com>
Signed-off-by: Yang Shen <shenyang39@huawei.com>
Reviewed-by: Xiu Jianfeng <xiujianfeng@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/crypto/hisilicon/qm.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/crypto/hisilicon/qm.c b/drivers/crypto/hisilicon/qm.c
index d9bdb433bef9..95d61fd62074 100644
--- a/drivers/crypto/hisilicon/qm.c
+++ b/drivers/crypto/hisilicon/qm.c
@@ -3073,7 +3073,6 @@ static void hisi_qm_uacce_put_queue(struct uacce_queue *q)
 {
 	struct hisi_qp *qp = q->priv;
 
-	hisi_qm_cache_wb(qp->qm);
 	hisi_qm_release_qp(qp);
 }
 
-- 
2.26.2

